src/hb-ot-name-table.hh

```diff
@@ -153,10 +153,23 @@ struct NameRecord
     auto *out = c->embed (this);
     if (unlikely (!out)) return_trace (nullptr);
 #ifdef HB_EXPERIMENTAL_API
-    if (name_table_overrides->has (nameID))
+    hb_vector_t<unsigned> record_ids;
+    if (!name_table_overrides->is_empty ())
     {
-      hb_bytes_t name_bytes = name_table_overrides->get (nameID);
-      unsigned text_size = hb_ot_name_convert_utf<hb_utf8_t, hb_utf16_be_t> (name_bytes, nullptr, nullptr);
+      if (unlikely (!record_ids.alloc (4)))
+      {
+        c->revert (snap);
+        return_trace (nullptr);
+      }
+      record_ids.push (platformID);
+      record_ids.push (encodingID);
+      record_ids.push (languageID);
+      record_ids.push (nameID);
```

Comment:
Why use a vector instead of a struct?!

Reply:
oh yes it was a struct initially and I don't remember why I changed it.... I'll make another try.

Reply:
oh... I remember now, after I added struct definition into `hb-ot-name-table.hh` and then included it in `hb-subset-plan.hh`, build failed because there's circular dependency: `hb-subset-plan.hh->hb-ot-name-table.hh->hb-open-type.hh->hb-subset.hh`, and `hb-subset.hh->hb-subset-plan.hh`, would you have any suggestion on where should I declare this struct? Thanks!

Reply:
or maybe I just create a new .hh file

Reply:
I would say in hb-subset-plan.hh should work. Just before the API definition. No?

Reply:
Can I put it in `hb-subset-input.hh`? The struct needs to be visible to hb-subset-input as well, and hb-subset-plan.hh depends on hb-subset-input.hh, I tried that it's

Reply:
Sure.